libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -104,6 +104,56 @@ string YulUtilFunctions::copyToMemoryFunction(bool _fromCalldata)
 	});
 }
 
+string YulUtilFunctions::requireOrAssertFunction(bool _assert, Type const* _messageType)
+{
+	string functionName =
+		string(_assert ? "do_assert" : "require") +
+		(_messageType ? "_withMessage" : "");
+
+	solAssert(_assert && !_messageType, "Asserts can't have messages!");
```

Comment:
Shouldn't this be `!assert || !_messageType`?
```suggestion
	solAssert(_assert && !_messageType, "Asserts can't have messages!");
```

Reply:
This is indeed wrong and will be fixed with the next push